### PR TITLE
NCTL: missed upgrade scenarios

### PR DESCRIPTION
**Pull request by TomVasile**
### Changes:
- Adds upgrade scenarios 4-7
- Refactor of NCTL upgrade wrapper for CI to allow using upgrade scenarios in nightly run
- Adds upgrade scenarios 4-7 to nightly runs
- Adds new wrapper `upgrade_from_stage_single_node.sh` for upgrading nodes individually
- Adds `|| true` to supervisorctl command that recently switched to exiting 1 when nothing was available to shutdown
- Adapts `await_n_eras` & `await_n_blocks` functions to optionally specify which node to query for consistency in tests

### Scenario(s) TL;DR
- 10 nodes running
- 8 upgrade, 2 left behind stalled
- 2 left behind are later staged
- 2 nodes restarted
- verify sucessful restart and syncing
- NOTE: All 4 tests use this generally with just small modifications
    - 2 binary upgrade scenarios
	- 2 protocol upgrade scenarios

### Tickets:
- https://app.zenhub.com/workspaces/cn-sre-60d363546afb85000e491ff6/issues/casper-network/sre/112
- https://app.zenhub.com/workspaces/engineering-60953fafb1945f0011a3592d/issues/casper-network/casper-node/1863
- https://app.zenhub.com/workspaces/engineering-60953fafb1945f0011a3592d/issues/casper-network/casper-node/1864

### Blocked by outcome(s) of:
- https://app.zenhub.com/workspaces/engineering-60953fafb1945f0011a3592d/issues/casper-network/casper-node/2087
- https://app.zenhub.com/workspaces/engineering-60953fafb1945f0011a3592d/issues/casper-network/governance/69

### CODE REVIEW

**Comment by TomVasile** [comment]
bors try

**Comment by TomVasile** [comment]
bors try

**Review by rob-casper — APPROVED** [review]

**Comment by TomVasile** [comment]
bors try

File: utils/nctl/sh/assets/upgrade_from_stage_single_node.sh [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+            "cfg['core']['validator_slots']=$COUNT_NODES;"
+            "toml.dump(cfg, open('$PATH_TO_CHAINSPEC', 'w'));"
+        )
+    fi
+
+    python3 -c "${SCRIPT[*]}"
+}
+
+#######################################
+# Sets node confgiuration files.
```
Comment (rafal-ch) [inline utils/nctl/sh/assets/upgrade_from_stage_single_node.sh]:
```suggestion
# Sets node configuration files.
```

Comment (TomVasile) [reply utils/nctl/sh/assets/upgrade_from_stage_single_node.sh]: https://github.com/casper-network/casper-node/pull/2106/commits/a3508fec8841c303df94e8f256622fcc765a1245

**Comment by TomVasile** [comment]
bors try

**Pull request merged**